lib/toy_robot_elixir/cli.ex

```diff
@@ -0,0 +1,58 @@
+defmodule ToyRobotElixir.Cli do
+  alias ToyRobotElixir.Robot
+  alias ToyRobotElixir.Table
+  alias ToyRobotElixir.Actions.Action
+
+  def welcome do
+    IO.puts(
+      "#{IO.ANSI.blue()}  #{IO.ANSI.magenta()}   Welcome to Toy Robot Elixir   #{IO.ANSI.blue()}  #{IO.ANSI.reset()}\n"
+    )
+
+    IO.puts("#{IO.ANSI.magenta()}Commands: #{IO.ANSI.blue()}")
+    IO.puts(" place x, y, face")
+    IO.puts(" move")
+    IO.puts(" left")
+    IO.puts(" right")
+    IO.puts(" report #{IO.ANSI.reset()}")
+  end
+
+  @spec place(x :: integer() | nil, y :: integer() | nil, face :: String.t() | atom() | nil) ::
+          {:ok, %Robot{}} | {:error, String.t()}
+  def place(x \\ nil, y \\ nil, face \\ nil)
```

Comment:
These are the commands that the user interacts with in the CLI, they simply call the Action module with the correct action with the standard table.